cabal-install/Distribution/Client/Types.hs

```diff
@@ -236,7 +236,7 @@ data PackageSpecifier pkg =
      -- | A fully specified source package.
      --
    | SpecificSourcePackage pkg
-  deriving (Eq, Show, Generic)
+  deriving (Eq, Show, Functor, Generic)
```

Comment:
I'd rather prefer if we used an explicit lambda with case analysis or a `mapSpecificSourcePackage` helper here. Code like `fmap fmap` is not super readable.